R/ops-addition.r

```diff
@@ -43,33 +43,28 @@ add_period_to_period <- function(per2, per1) {
 
 add_period_to_date <- function(per, date) {
 
-  lt <- as.POSIXlt(date)
+  new <- update(date,
```

Comment:
You have replaced 1 update call with 3 and introduced quite a number of POSIXct -> POSIXlt conversions by dropping the original `lt <- ` line. This will surely have a dire consequence on speed.

Why do you need so many changes if  you can just check if output is divisible by 86400 as @hadley proposed in the other thread?